EIPS/eip-remove_utf8_support.md

```diff
@@ -0,0 +1,140 @@
+
+##### length
+
+The length is the length of the data in bytes (same as it is now).
+
+##### range
+
+The range is an unsigned 8 byte integer, which is to be interpreted as the upper bound for character values. The lower bound is always `0`.
+
+A range of `0` means that the standard byte encoding is used, so it is the same as setting it to `255`. This is to make `string` compatible with `bytes`.
+
+Validating a string is done by checking that none of its characters has a value that is outside of the range.
+
+Note that non power-of-two character-widths are not ideal, because they lead to inefficient use of storage (a single value could have its bytes spread out over more then one storage slot).
+
+##### reserved
+
+Reserved space can be used to store additional information. Using it is optional, but it can not be assumed to be empty.
+
+#### Using the reserved section
+
+The reserved section could be used to make more advanced strings possible.
+
+| Section              | position (little endian)  |  size  | total size |
+|----------------------|---------------------------|--------|------------|
+| length (bytes)       | 0                         | 8      | 8          |
+| range                | 8                         | 8      | 16         |
+| format               | 16                        | 8      | 24         |
+| reserved             | 24                        | 7      | 31         |
+| extra bytes          | 31                        | 1      | 32         |
+
+##### format
+
+The format is an identifier, and would usually indicate that the string has a particular format. This format could for example be backed by a *character mapping*, but that is not a requirement (see the character mappings section for more information).
+
+The zero key (i.e. `key == 0`) would be reserved for the standard byte encoding (range `0`).
+
+##### extra bytes (optional)
+
+The extra bytes field is optional, but can (and should) be used to avoid having to recompute the character width every time it is needed. The value held in this field must be one less then the (smallest) amount of bytes that is required to hold a character. Putting it in the last byte means it can be accessed using a single bit-shift operation (no masking is needed), which means that index access and length computation becomes slightly more efficient.
+
+The reason for using extra bytes and not the character width itself is to stay compatible with the standard byte encoding.
+
+### Character mappings and keys
+
+A character mapping can be seen as a *locale* where the alphabet of the string is defined. The alphabet must be an array in which every index is mapped to a (unique) value from some set of integer values (UTF-8 code-points for example).
+
+An example alphabet would be: `[0, 'a', 'b', 'c', 'd', 'e', 'f', 'g']`. This contains 8 UTF-8 characters indexed from 0 to 7. A string created from this alphabet would require 8 bits per character, so the range for this alphabet is `7`.
```

Comment:
Choice (a). The range is 7 because the available values are 0, 1, 2, 3, 4, 5, 6, 7.

The idea was just a simple example of how another encoding could be used, and how strings could work very efficiently without knowing anything about the actual alphabet that is used. That is essentially abstracted away in this model. If the range is `N`, then automatically every number from `0` to `N` is assumed to be valid, according to the protocol. It is then trivial to prove that a character is valid in a contract - you just have to check whether `c < N` for all characters. 

In practice, you would create a mapping from an alphabet (made of `N+1` characters) to the range [0, N] off-chain, where each character corresponds to a unique number. If you first check the validity of a string (let's say it's utf-8 encoded), then map it to the given range and pass it to a contract (using this encoding), then the range check would be equivalent to the off chain check, i.e. if a character passes the check off chain it would pass on-chain as well, and the same if it doesn't pass.

It would also mean sub strings and all other type of composition/decomposition would only require trivial processes, unlike with UTF-8 strings, which requires either pre-indexing, or processes that can not be done in constant time.

I realized that this could be useful, but it does not seem popular and will probably create more problems then it solves.